### ISSUE TITLE
[Paper Note] What do Language Models Learn and When? The Implicit Curriculum Hypothesis, Emmy Liu+, arXiv'26, 2026.04

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#5590]
# URL
- https://arxiv.org/abs/2604.08510
# Authors
  - Emmy Liu
  - Kaiser Sun
  - Millicent Li
  - Isabelle Lee
  - Lindia Tjuatja
  - Jen-tse Huang
  - Graham Neubig
# Abstract
  - Large language models (LLMs) can perform remarkably complex tasks, yet the fine-grained details of how these capabilities emerge during pretraining remain poorly understood. Scaling laws on validation loss tell us how much a model improves with additional compute, but not what skills it acquires in which order. To remedy this, we propose the Implicit Curriculum Hypothesis: pretraining follows a compositional and predictable curriculum across models and data mixtures. We test this by designing a suite of simple, composable tasks spanning retrieval, morphological transformations, coreference, logical reasoning, and mathematics. Using these tasks, we track emergence points across four model families spanning sizes from 410M-13B parameters. We find that emergence orderings of when models reach fixed accuracy thresholds are strikingly consistent ($ρ= .81$ across 45 model pairs), and that composite tasks most often emerge after their component tasks. Furthermore, we find that this structure is encoded in model representations: tasks with similar function vector representations also tend to follow similar trajectories in training. By using the space of representations derived from our task set, we can effectively predict the training trajectories of simple held-out compositional tasks throughout the course of pretraining ($R^2 = .68$-$.84$ across models) without previously evaluating them. Together, these results suggest that pretraining is more structured than loss curves reveal: skills emerge in a compositional order that is consistent across models and readable from their internals.
# Translation (by gpt-5-nano)
- 大規模言語モデル（LLMs）は非常に複雑なタスクを実行できる一方で、これらの能力が事前学習中にどのように細かな段階で現れるかという細部はまだよく理解されていません。検証損失のスケーリング則は、追加計算によってモデルがどれだけ改善するかを教えてくれますが、どのスキルをどの順序で獲得するかは教えてくれません。これを是正するために、暗黙のカリキュラム仮説を提案します：事前学習はモデル間およびデータ混合を横断して構成的で予測可能なカリキュラムに従います。これを検証するために、検索、形態変換、共参照、論理推論、数学を横断する、シンプルで組み合わせ可能なタスクのセットを設計しました。これらのタスクを用いて、410M〜13Bパラメータのサイズにまたがる4つのモデルファミリーにおける出現点を追跡します。出現の順序は、モデルが固定の精度閾値に到達する時期の並びが驚くほど一貫しており（ρ=0.81、45組のモデル対で）、複合タスクは多くの場合、それを構成するタスクの後に現れることが分かりました。さらに、この構造はモデルの表現に組み込まれていることが分かります。類似の機能ベクトル表現を持つタスクは、訓練中にも類似した軌道をたどる傾向があります。私たちのタスク集合から導出した表現空間を用いることで、事前学習の過程全体を通じ、単純な留出型の組成タスクの訓練経路を、これまで評価せずとも効果的に予測できます（R^2 = 0.68〜0.84、モデル間）。これらを総合すると、事前学習は損失曲線が示すよりも構造化されていることを示唆しており、スキルはモデル間で一貫性のある組成的順序で現れ、内部から読み取り可能です。
# Summary (by gpt-4o-mini)
- 大規模言語モデル（LLMs）の事前学習におけるスキル獲得の順序を理解するための「暗黙のカリキュラム仮説」を提案。シンプルかつ組み合わせ可能なタスクを用い、モデル間の一貫した出現順序を追跡。特定のパラメータ範囲で構成的なタスクが後に現れる傾向があり、モデルの表現に組み込まれていることを示す。予測可能な訓練経路を通じて、事前学習は構造化されていると示唆。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/gneubig/status/2057134088494977393?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
これは、著者ポストしっかり読みたい

**Comment by AkihikoWatanabe** [comment]
- モデルファミリー・DataMixtureにはよらず、事前学習では構成的で、かつ予測可能なカリキュラムに則って学習が進行し、かつモデルの内部状態から各スキルがどのように学習されていくかを予測できるという仮説を立て、
- この仮説を検証するために、91種類の構成的なタスクを定義し、emergence（=当該タスクの性能が閾値を超えること）を4種類のモデルファミリーにおける9つのモデル、様々なDataMixtureの元で追跡した。タスクの例は以下:
  - simple tasks: 文字列操作/形態素の変換/知識の抽出/翻訳など
  - composite tasks: 複数の基礎的な操作のsequentialな組み合わせによって実現されるタスク
    - たとえば、`gerund_upper` は大文字への変換➡︎動名詞への変換という順番で定義される。

<img width="672" height="478" alt="Image" src="https://github.com/user-attachments/assets/fb798ada-9190-4c08-aa02-16dbe0aedb12" />

- 様々なモデルファミリーをテストしたところ、LLMは事前学習の間におおむね（完璧ではないが）同じ順番でスキルを獲得していくことが明らかになった
  - たとえば、Figure 1を見ると、性能の伸び方は異なるものの、閾値を50%としたときのemergenceの順番はモデルの間で一貫していることがわかる。Table2も参照のこと。

<img width="1087" height="468" alt="Image" src="https://github.com/user-attachments/assets/c4a8073f-2ad9-488b-8f37-15fd651c7936" />

- composite tasksは、それらのタスクの構成要素が獲得された後にemergeすることが明らかになった（54/76ケース）
  - 例外的に、composition taskが構成要素よりも先に習得されたものが3例ほど存在した
- また、あるcomposite taskの学習曲線を、類似したFunction Vectors [^1] を持つcomposite taskから予測できるか？（i.e., 類似したタスクは同じような学習曲線を持つか？）を検証。
- これを実施するために、composite taskに対してleave-one-outを実施し、類似したタスクのFunction Vectorsから学習の軌跡を予測できるかを実験したところ、R^2スコアが0.68--0.84程度の性能で予測することができた。
  - Function Vectors: #5669

<img width="823" height="566" alt="Image" src="https://github.com/user-attachments/assets/171547dd-7633-45e4-bac9-14c4d6a305eb" />

[^1]: Function Vectorsとは、LLMに遂行させるタスクのinput-outputの変換の関係性を保持し、タスクを遂行させる際にLLMに対して強い影響力を持つ内部のactivationsのことを指す。
